File: android/src/main/java/com/imba97/plugins/audio/stream/AudioStreamPlugin.java
package com.imba97.plugins.audio.stream;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;
import com.getcapacitor.annotation.Permission;

@CapacitorPlugin(
    name = "AudioStream",
    permissions = {
        @Permission(
            alias = "microphone",
            strings = { android.Manifest.permission.RECORD_AUDIO }
        )
    }
)
public class AudioStreamPlugin extends Plugin {

    private AudioStream audioStream;

    @Override
    public void load() {
        super.load();
        audioStream = new AudioStream();
        audioStream.setBridge(this.bridge); // 需要给 AudioStream 添加 setBridge 方法
    }

    @PluginMethod
    public void start(PluginCall call) {
        try {
            audioStream.startRecording();
            call.resolve();
        } catch (Exception e) {
            call.reject("Failed to start recording", e);
        }
    }

    @PluginMethod
    public void stop(PluginCall call) {
        try {
            audioStream.stopRecording();
            call.resolve();
        } catch (Exception e) {
            call.reject("Failed to stop recording", e);
        }
    }
}

File: ios/Sources/AudioStreamPlugin/AudioStreamPlugin.swift
import Foundation
import Capacitor
import AVFoundation

@objc(AudioStreamPlugin)
public class AudioStreamPlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "AudioStreamPlugin"
    public let jsName = "AudioStream"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "start", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "stop", returnType: CAPPluginReturnPromise)
    ]

    private var audioStream: AudioStream!

    override public func load() {
        super.load()
        audioStream = AudioStream()
        audioStream.setPlugin(plugin: self)
    }

    @objc func start(_ call: CAPPluginCall) {
        do {
            try audioStream.startRecording()
            call.resolve()
        } catch {
            call.reject("Failed to start recording: \(error.localizedDescription)")
        }
    }

    @objc func stop(_ call: CAPPluginCall) {
        audioStream.stopRecording()
        call.resolve()
    }
}
